src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildLazyLoadTest.cs

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+using static Microsoft.AspNetCore.Components.WebAssembly.Build.WebAssemblyRuntimePackage;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+{
+    public class BuildLazyLoadTest
+    {
+        [Fact]
+        public async Task Build_LazyLoadExplicitAssembly_Debug_Works()
+        {
+            // Arrange
+            using var project = ProjectDirectory.Create("standalone", additionalProjects: new[] { "razorclasslibrary", "rclwithnodependencies" });
+            project.Configuration = "Debug";
+
+            project.AddProjectFileContent(
+@"
+<ItemGroup>
+    <ProjectReference Include='..\rclwithnodependencies\RclWithNoDeps.csproj'/>
+</ItemGroup>
+<ItemGroup>
+    <BlazorLazyLoad Include='RclWithNoDeps.dll' />
+</ItemGroup>");
+
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, args: "/restore");
+
+            var buildOutputDirectory = project.BuildOutputDirectory;
+
+            // Verify that a blazor.boot.json file has been created
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "blazor.boot.json");
+            // And that the assembly is in the output
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "RclWithNoDeps.dll");
+
+            var bootJson = ReadBootJsonData(result, Path.Combine(buildOutputDirectory, "wwwroot", "_framework", "blazor.boot.json"));
+
+            // And that it has been labelled as a dynamic assembly in the boot.json
+            var dynamicAssemblies = bootJson.resources.dynamicAssembly;
+            Assert.NotNull(dynamicAssemblies);
+            Assert.Contains("RclWithNoDeps.dll", dynamicAssemblies.Keys);
```

Comment:
We should verify `RclWithNoDeps.dll` (and the subset of assemblies that only appear as it's dependencies) do not appear in `bootJson.resources.assembly`.

Reply:
Added this.